### ISSUE TITLE
Custom loss functions in keras

### ISSUE THREAD

**Issue by Sadegh-Saberian** [keras-team/keras#10885]
Hi,

I was looking at [this](https://arxiv.org/pdf/1602.02644.pdf) paper, but i get confused on how they computed their loss function. they mentioned, "we compute distances between image features extracted by deep neural networks." what i dont understand here is that how exactly did they managed to do this in implementation. more specifically, consider that the deep neural network they used for feature extraction is a black box function called F. in the loss function, they have to calculate this function for both the image and its reconstruction. however, how do they manage to calculate the gradient for the back propagation phase if the F is assumed to be a black box function?

I would really appreciate if someone could help me understand this.

**Comment by Dref360** [comment]
Hi!
This issue isn't related to a bug/enhancement/feature request or other accepted types of issue.

To ask questions, please see the following resources : 
* https://keras.io/
* https://keras.io/#support
* https://gitter.im/Keras-io/Lobby

Thanks!

If you think I made a mistake, please open another issue.
